refresh.template.py

```diff
@@ -761,6 +806,40 @@ def _get_cpp_command_for_files(compile_action):
     return source_files, header_files, compile_action.arguments
 
 
+def _get_swift_command_for_files(compile_action):
+    """Reformat compile_action into a compile command sourcekit-lsp (https://github.com/apple/sourcekit-lsp) can understand.
+
+    Compile_action was produced by rule_swift (https://github.com/bazelbuild/rules_swift)
+    """
+    # Patch command
+    compile_action.arguments = _all_platform_patch(compile_action.arguments)
+    compile_action.arguments = _apple_swift_patch(compile_action.arguments)
+    compile_action.arguments = _apple_platform_patch(compile_action.arguments, compile_action.environmentVariables)
+
+    # Source files is end with `.swift`
+    source_files = set(filter(lambda x: x[-6:] == ".swift", compile_action.arguments))
```

Comment:
Do we expect multiple--or files without a swift extension?
(Another good spot for endswith if you want.)

Reply:
This is a compile command argument sample and it seems that the only way to identify the source is ending with .swift

```
"arguments": [
      "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc",
      "-target",
      "arm64-apple-macos16.1",
      "-sdk",
      "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk",
      "-emit-object",
      "-output-file-map",
      "bazel-out/darwin_arm64-dbg/bin/app/home/home.output_file_map.json",
      "-emit-module-path",
      "bazel-out/darwin_arm64-dbg/bin/app/home/app_home_home.swiftmodule",
      "-enable-bare-slash-regex",
      "-DDEBUG",
      "-Onone",
      "-Xfrontend",
      "-serialize-debugging-options",
      "-enable-testing",
      "-g",
      "-debug-prefix-map",
      "/Users/xinzhengzhang/Workspace/rules_gripper/examples/iosapps=.",
      "-Ibazel-out/darwin_arm64-dbg/bin/services/app/home",
      "-Ibazel-out/darwin_arm64-dbg/bin/services/base/router",
      "-Ibazel-out/darwin_arm64-dbg/bin/services/app/live/liveService",
      "-Ibazel-out/darwin_arm64-dbg/bin/external/build_bazel_rules_gripper/GPModule",
      "-Xcc",
      "-iquote.",
      "-Xcc",
      "-iquotebazel-out/darwin_arm64-dbg/bin",
      "-Xfrontend",
      "-color-diagnostics",
      "-enable-batch-mode",
      "-module-name",
      "app_home_home",
      "-parse-as-library",
      "-static",
      "-Xcc",
      "-O0",
      "-Xcc",
      "-DDEBUG=1",
      "-Xcc",
      "-fstack-protector",
      "-Xcc",
      "-fstack-protector-all",
      "app/home/home.swift"
    ],
```